sunpy/data/data_manager/manager.py

```diff
@@ -172,10 +181,30 @@ def get(self, name):
         `KeyError`
             If ``name`` is not in the cache.
         """
-        return pathlib.Path(self._file_cache[name])
+        return pathlib.Path(self._file_cache[name][self._namespace])
 
     def _cache_has_file(self, urls):
         for url in urls:
             if self._cache._get_by_url(url):
                 return True
         return False
+
+    def _get_module(self, func):
+        """
+        Returns the name of the module (appended with a dot) that a function belongs to.
+        This is made into a separate function for the ability to patch in tests.
+
+        Parameters
+        ----------
+        func : `function`
+            A function who's module is to be found.
+
+        Returns
+        -------
+        `module`:
+            A `str` that represents the module name appended with a dot.
+        """
+        module = func.__module__.lstrip('.').split('.')[0] + '.'
+        if module == '__main__.':
```

Comment:
Is there a test for this case?

Reply:
No, writing such a test is similar to the problem I faced before, where I was not able to fake the module name of a function for the tests. I dabbled in `sys.module` but with no luck. I think this has got something to do with `pytest`, but I'm not sure. I ended up patching the `_get_module` function instead.

Do you have any suggestions on how we could do this?

Reply:
Nope.